source/grid/grid_generator.cc

```diff
@@ -28,11 +30,1180 @@
+                    double y_u = y_c + y_t * std::cos(theta);
+                    naca_points.push_back({x_u, y_u});
+                  }
+                else
+                  {
+                    double x_l = x + y_t * std::sin(theta);
+                    double y_l = y_c - y_t * std::cos(theta);
+                    naca_points.push_back({x_l, y_l});
+                  }
+              }
+          }
+        return naca_points;
+      }
+
+
+
+      /**
+       * Set airfoil length (i.e. chord length) to a desired length.
+       * calls function set_airfoil_length() for each vector of the array
+       *
+       * @param array containing upper and lower vector
+       * @param desired length
+       * @return array of two vectors with desired length
+       */
+      static std::array<std::vector<Point<2>>, 2>
+      set_airfoil_length(const std::array<std::vector<Point<2>>, 2> &input,
+                         const double desired_len)
+      {
+        std::array<std::vector<Point<2>>, 2> output;
+        output[0] = set_airfoil_length(input[0], desired_len);
+        output[1] = set_airfoil_length(input[1], desired_len);
+
+        return output;
+      }
+
+      /**
+       * Set airfoil length (i.e. chord length) to a desired length.
+       *
+       * @param inputvector
+       * @param desired length
+       * @return scaled vector with desired length
+       * */
+      static std::vector<Point<2>>
+      set_airfoil_length(const std::vector<Point<2>> &input,
+                         const double                 desired_len)
+      {
+        std::vector<Point<2>> output(input.size());
+
+        const auto scale = desired_len / input.front().distance(input.back());
+
+        for (unsigned i = 0; i < input.size(); i++)
+          output[i] = input[i] * scale;
+
+        return output;
+      }
+
+      /**
+       * Interpolation among non_equidistant_points in order to obtain
+       * (number_points)-equidistant points
+       *
+       * @param non_equidistan_points: vector containing non equidistan points
+       * @param number_points: indicating desired amount of equidistant points
+       * @return equidist: vector containing (number_points) equidistant points
+       */
+      static std::vector<Point<2>>
+      make_points_equidistant(
+        const std::vector<Point<2>> &non_equidistant_points,
+        const unsigned int           number_points)
+      {
+        const unsigned int n_points =
+          non_equidistant_points
+            .size(); // number provided airfoilpoints to interpolate
+
+        // calculate arclength
+        std::vector<double> arclength_L(non_equidistant_points.size(), 0);
+        for (unsigned int i = 0; i < non_equidistant_points.size() - 1; i++)
+          arclength_L[i + 1] =
+            arclength_L[i] +
+            non_equidistant_points[i + 1].distance(non_equidistant_points[i]);
+
+
+        double airfoil_length =
```

Comment:
```suggestion
        const auto airfoil_length =
```

Reply:
how about parameters `deltaX` and `l` in the same function, can't they be constant as well?
`const double deltaX` and `const double l`

Reply:
Make as much as possible `const`.

Reply:
Statements at one place is true for all other places...

Reply:
in this function no more parameters can be const right? Sorry, but I don't understand what you mean by `Statements at one place is true for all other places...`